cpp/src/arrow/array.h

```diff
@@ -734,6 +745,27 @@ class ARROW_EXPORT DictionaryArray : public Array {
  private:
   void SetData(const std::shared_ptr<ArrayData>& data);
 
+  /// \brief Check if all indices are within valid range
+  ///
+  /// \param[in] indices dictionary indices
+  /// \param[in] range valid range of indices (0 <= index < range)
+  template <typename ArrowType>
+    static bool SanityCheck(const std::shared_ptr<Array>& indices, const int64_t range) {
```

Comment:
Formatting looks off by 2 spaces here.

Reply:
The variable name `range` is a bit imprecise. Perhaps `upper_bound`?